sim/src/main.rs

```diff
@@ -233,8 +234,9 @@ impl RunStatus {
         }
 
         let mut bad = 0;
-        // Let's try an image halfway through.
-        for i in 1 .. total_count {
+        let mut stops: Vec<i32> = (1..total_count).collect();
```

Comment:
Actually, thinking about this, this change doesn't really have any affect. The 'flash' is not passed mutably, so is restored to the initial state after each call to try_upgrade. It doesn't really matter what order the individual tests are run in. Did you find problems this way?